middle_end/flambda/inline_and_simplify.ml

```diff
@@ -856,16 +868,18 @@ and simplify_over_application env r ~args ~args_approxs ~function_decls
   let full_app_approxs, _ =
     Misc.Stdlib.List.split_at arity args_approxs
   in
+  (* FIXME insert Tail appropriately if position = Apply_tail *)
   let expr, r =
     simplify_full_application env r ~function_decls ~lhs_of_application
       ~closure_id_being_applied ~function_decl ~value_set_of_closures
-      ~args:full_app_args ~args_approxs:full_app_approxs ~dbg
+      ~args:full_app_args ~args_approxs:full_app_approxs ~dbg ~position
```

Comment:
Once `Tail` is inserted, as mentioned in the FIXME, this application and the one below should have position `Apply_nontail`.